src/lib/common/sol-certificate-impl-linux.c

```diff
@@ -147,3 +148,24 @@ sol_cert_get_filename(const struct sol_cert *cert)
 
     return cert->filename;
 }
+
+SOL_API struct sol_blob *
+sol_cert_get_contents(const struct sol_cert *cert)
+{
+    struct sol_blob *blob;
+    char *data;
+
+    SOL_NULL_CHECK(cert, NULL);
+
+    data = sol_util_load_file_string(cert->filename, NULL);
```

Comment:
Why don't you use second parameter to get `data` size instead of doing a `strlen`?